lib/common/common.go

```diff
@@ -112,3 +114,143 @@ func ValidateLayerId(id string) error {
 	}
 	return nil
 }
+
+/*
+ * Media Type Selectors Section
+ */
+
+const (
+	MediaTypeDockerV21Manifest       = "application/vnd.docker.distribution.manifest.v1+json"
+	MediaTypeDockerV21SignedManifest = "application/vnd.docker.distribution.manifest.v1+prettyjws"
+	MediaTypeDockerV21ManifestLayer  = "application/vnd.docker.container.image.rootfs.diff+x-gtar"
+
+	MediaTypeDockerV22Manifest     = "application/vnd.docker.distribution.manifest.v2+json"
+	MediaTypeDockerV22ManifestList = "application/vnd.docker.distribution.manifest.list.v2+json"
+	MediaTypeDockerV22Config       = "application/vnd.docker.container.image.v1+json"
+	MediaTypeDockerV22RootFS       = "application/vnd.docker.image.rootfs.diff.tar.gzip"
+
+	MediaTypeOCIManifest     = spec.MediaTypeImageManifest
+	MediaTypeOCIManifestList = spec.MediaTypeImageManifestList
+	MediaTypeOCIConfig       = spec.MediaTypeImageConfig
+	MediaTypeOCILayer        = spec.MediaTypeImageLayer
+)
+
+// MediaTypeOption represents the media types for a given docker image (or oci)
+// spec.
+type MediaTypeOption int
+
+const (
+	MediaTypeOptionDockerV21 = iota
+	MediaTypeOptionDockerV22
+	MediaTypeOptionOCIV1Pre
+)
+
+// MediaTypeSet represents a set of media types which docker2aci is to use when
```

Comment:
You're right in that nothing particularly bad will happen if the set invariant is violated, but it would probably be best for me to mention it regardless.